lib/s3.py

```diff
@@ -107,13 +109,19 @@ def urlopen(url: urllib.parse.ParseResult, method='GET', headers={}, data=b'') -
     """Same as sign_request() but calls urlopen() on the result"""
     headers = sign_request(url, method, headers, hashlib.sha256(data).hexdigest())
     request = urllib.request.Request(url.geturl(), headers=headers, method=method, data=data)
+
+    # is this a redhat store?
+    if any([s for s in REDHAT_STORES if url.netloc in s]):
```

Comment:
1. This doesn't need to be a list comprehension.  A generator is better here.
2. Just put the condition at the front.

Like this:

```
    if any(url.netloc in store for store in REDHAT_STORES):
        ....
```

Reply:
This would also make it absolutely trivial to add to the `curl` helper, if you wanted to go that route (which might be useful for the CLI interface of `s3.py` still).

Reply:
Cleaned this up.